python/ray/data/read_api.py

```diff
@@ -577,9 +577,7 @@ def read_csv(
     ray_remote_args: Dict[str, Any] = None,
     arrow_open_stream_args: Optional[Dict[str, Any]] = None,
     meta_provider: BaseFileMetadataProvider = DefaultFileMetadataProvider(),
-    partition_filter: Optional[
-        PathPartitionFilter
-    ] = CSVDatasource.file_extension_filter(),
```

Comment:
Is the same logic applicable for other data types (e.g. json, parquet)?

Reply:
> Is the same logic applicable for other data types (e.g. json, parquet)?

We don't have default filtering for Parquet. So Parquet is good.
We have default filter for JSON to filter out files wihtout `.json` extension. Just tried out Arrow and Spark on my laptop and they won't filter out files when reading JSON. We can change the behavior for JSON files as well later in another followup PR. To me, CSV fix is more urgent as there're multiple user reports (but it can also be the case that our `read_json` is not popular).

Reply:
Oh interesting, `read_parquet` doesn't have default filtering while `read_parquet_bulk` does (which seems a bit unintuitive to me, since `read_parquet_bulk` doesn't support directories).

In that case I think it's reasonable to go forward with this for now and follow-up with a more holistic/consistent solution.

One question - is the error message when reading a non-CSV file directly actionable to the user, especially for users who previously relied on this default behavior? E.g. when the file is a TSV, or if the file is some random file that should be excluded.

Reply:
> Oh interesting, read_parquet doesn't have default filtering while read_parquet_bulk does (which seems a bit unintuitive to me, since read_parquet_bulk doesn't support directories).

Yeah, this looks unintuitive to me too. Don't think we should have different between these two Parquet APIs.

> One question - is the error message when reading a non-CSV file directly actionable to the user, especially for users who previously relied on this default behavior? E.g. when the file is a TSV, or if the file is some random file that should be excluded.

Yeah agreed, we should have an actionable error message. That's exactly I am doing now, plan to have another PR to tackle the error message when reading non-CSV files by mistake, and it should also apply when reading a malformed `.csv` file.

Reply:
Added handling to provide more detailed error message. Added an example error message when reading non-CSV file in PR description.